spec/acceptance/upserting_resource_monitor_spec.rb

```diff
@@ -0,0 +1,75 @@
+require "rails_helper"
+# rubocop:disable BlockLength
+RSpec.describe "Create an enabled monitor" do
+  subject do
+    LinkMonitor::UpsertResourceMonitor.new(links: links, service: service, resource_type: "Text", resource_id: 1).call
+  end
+
+  let(:links) { ["http://example.com/a", "http://example.com/b"] }
+  let(:service) { "local-link-manager" }
+
+  context "without links" do
+    let(:links) { [] }
+    it { is_expected.to be_a(ResourceMonitor) }
+  end
+
+  context "with links" do
+    it { is_expected.to be_a(ResourceMonitor) }
+
+    it "should persist data" do
+      monitor = ResourceMonitor.find(subject.id)
+
+      expect(monitor.links).not_to be_empty
+      expect(monitor.monitor_links).not_to be_empty
+    end
+  end
+
+  context "when another monitor exists" do
+    before do
+      LinkMonitor::UpsertResourceMonitor.new(links: links, service: "gov-uk", resource_type: "Text", resource_id: 2).call
+      subject
+    end
+
+    it "does not duplicate links" do
+      expect(Link.all.count).to eq(2)
+    end
+
+    it "does create new monitor links" do
+      expect(MonitorLink.all.count).to eq(4)
+    end
+  end
+
+  context "when we create another resource monitor for an existing resource monitor" do
+    let(:resource) { subject }
+
+    before do
+      resource
+
+      links << "http://example.com/c"
+      links.shift
```

Comment:
This is removing `/a` from the links array but I reckon we could make this easier to comprehend.

This might be a little clearer:

``` ruby
links.delete("http://example.com/a")
```